planner/core/stats.go

```diff
@@ -431,8 +431,8 @@ func (ds *DataSource) DeriveStats(childStats []*property.StatsInfo, selfSchema *
 			}
 		}
 	}
-	cond, _ := ds.ctx.GetSessionVars().StmtCtx.GetCacheTable(ds.tableInfo.ID)
-	if isPossibleIdxMerge && sessionAndStmtPermission && needConsiderIndexMerge && isReadOnlyTxn && ds.tableInfo.TempTableType != model.TempTableLocal && !cond {
+	readFromTableCache := ds.ctx.GetSessionVars().StmtCtx.ReadFromTableCache
```

Comment:
if a statement involves two tables but only one of it is using table cache. Will index merge still be disabled for  another table?

Reply:
https://github.com/pingcap/tidb/issues/29682